src/main/java/org/jenkinsci/plugins/oic/OicServerWellKnownConfiguration.java

```diff
@@ -158,6 +161,56 @@ public OIDCProviderMetadata toProviderMetadata() {
         throw new IllegalStateException("Well known configuration could not be loaded, login can not preceed.");
     }
 
+    /**
+     * Filter {@link OIDCProviderMetadata} for all the JWS/JWE algorithms and  Encryption methods
+     */
+    protected static void filterNonCompliantAlgorithms(OIDCProviderMetadata oidcProviderMetadata) {
+
+        // Filter Jws Algorithms
+        List<List<JWSAlgorithm>> jwsAlgorithmLists = List.of(
+                Optional.ofNullable(oidcProviderMetadata.getIDTokenJWSAlgs()).orElse(List.of()),
+                Optional.ofNullable(oidcProviderMetadata.getTokenEndpointJWSAlgs())
+                        .orElse(List.of()),
+                Optional.ofNullable(oidcProviderMetadata.getIntrospectionEndpointJWSAlgs())
+                        .orElse(List.of()),
+                Optional.ofNullable(oidcProviderMetadata.getRevocationEndpointJWSAlgs())
+                        .orElse(List.of()),
+                Optional.ofNullable(oidcProviderMetadata.getRequestObjectJWSAlgs())
+                        .orElse(List.of()),
+                Optional.ofNullable(oidcProviderMetadata.getDPoPJWSAlgs()).orElse(List.of()),
+                Optional.ofNullable(oidcProviderMetadata.getAuthorizationJWSAlgs())
+                        .orElse(List.of()),
+                Optional.ofNullable(oidcProviderMetadata.getBackChannelAuthenticationRequestJWSAlgs())
+                        .orElse(List.of()));
+
```

Comment:
this does not make sense to me.
creating a list of lists, of different algortihms that are used for different functions.

am I missing the side effect of filtering in this method?